pkg/apis/pravega/v1beta1/pravega.go

```diff
@@ -383,6 +383,21 @@ func (s *PravegaSpec) withDefaults() (changed bool) {
 		s.Options = map[string]string{}
 	}
 
+	if s.Options["bookkeeper.ensemble.size"] == "" {
```

Comment:
@nishant-yt  since we are making all the 3 fields mandatory, will it enter this code flow

Reply:
@anishakj all the fields are not mandatory. If the user doesn't provide value for any of these options, we are setting it to `3` by default. 
One of instances when it will enter the above mentioned code flow is when the user doesn't provide any value for `bookkeeper.ensemble.size` `bookkeeper.write.quorum.size` and `bookkeeper.ack.quorum.size`